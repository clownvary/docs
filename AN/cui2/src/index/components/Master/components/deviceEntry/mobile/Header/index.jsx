import radium from 'radium';
import React, { Component } from 'react';
import { withRouter } from 'react-router';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { injectIntl } from 'react-intl';
import findIndex from 'lodash/findIndex';
import { Link } from 'shared/components/Link';
import decodeHtmlStr from 'react-base-ui/lib/utils/decodeHtmlStr';
import { Icon } from 'react-base-ui/lib/components/SVG';
import 'react-base-ui/lib/svgs';

import { showMenuAction } from 'index/components/Master/actions';
import { initCartPageAsyncAction } from 'index/modules/Cart/ShoppingCart/actions/common';
import selfMessages from 'shared/translation/messages/common';

import 'shared/assets/images/logo.png';

import reloadMyCart from '../../common/reloadMyCart';

import './index.less';

export class MobileHeader extends Component {

  static contextTypes = {
    configurations: PropTypes.object,
    systemSettings: PropTypes.object
  }

  static propTypes = {
    cartCount: PropTypes.number,
    showMenuAction: PropTypes.func,
    showMenu: PropTypes.bool
  }

  componentDidMount() {
    window.onpopstate = () => {
      this.props.showMenuAction(false);
    };
  }

  componentWillUnmount() {
    window.onpopstate = null;
  }

  onClickMyCart(myCartUrl) {
    this.props.showMenuAction(false);
    this.reloadMyCart(myCartUrl);
  }

  reloadMyCart = reloadMyCart;

  render() {
    const { showMenu, cartCount, intl: { messages } } = this.props;
    const { configurations, systemSettings } = this.context;
    const logo = systemSettings.getIn(['header', 'logo']);
    const menus = systemSettings.getIn(['navigation', 'menus']).toJS();
    const index = findIndex(menus, o => o.title === 'MyCart');
    const myCartUrl = menus[index].url;

    return (
      <div className="an-responsiveHeader">
        <header className="an-responsiveHeader__wrapper is-center">
          <div className="an-responsiveHeader__menu">
            <button
              onClick={() => { this.props.showMenuAction(); }}
              onTouchStart={() => { }}
              // add empty onTouchStart event  trigger safari :active style
            >
              <Icon name={showMenu ? 'remove' : 'bars'} aria-label="remove icon" />

              {messages[selfMessages.menuText.id]}
            </button>
          </div>
          <div className="an-responsiveHeader__logo">
            <a href={configurations.get('banner_logo_link')} target="_blank" rel="noopener noreferrer">
              { logo ?
                <img
                  src={logo.get('url')}
                  alt={`${decodeHtmlStr(logo.get('title'))}`}
                  title={decodeHtmlStr(logo.get('title'))}
                />
                : null
             }
            </a>
          </div>
          <div className="an-responsiveHeader__cart">
            <Link to={myCartUrl} onClick={() => this.onClickMyCart(myCartUrl)}>
              <button
                onTouchStart={() => { }}
                // add empty onTouchStart event  trigger safari :active style
              >
                <Icon
                  name="shopping-cart"
                  className="an-responsiveHeader__cart-icon"
                />
                {
                  cartCount ? <span className="an-responsiveHeader__cart-count">{cartCount}</span> : null
                }
                <span className="an-responsiveHeader__cart-empty-count">{messages[selfMessages.emptyText.id]}</span>
              </button>
            </Link>
          </div>
        </header>
      </div>
    );
  }
}

export default injectIntl(connect(
  state => ({
    cartCount: state.master.get('cartCount'),
    showMenu: state.master.get('showMenu')
  }), {
    showMenuAction, initCartPageAsyncAction
  }
)(radium(withRouter(MobileHeader))));
