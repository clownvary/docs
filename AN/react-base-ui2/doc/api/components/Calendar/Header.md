## Header 



### Prop Types
Property | Type | Default Value | Is Required | Description
:--- | :--- | :--- | :--- | :---
onPrevClick|func|&ensp;|true|&ensp;
onNextClick|func|&ensp;|true|&ensp;
onTitleClick|func|&ensp;|true|&ensp;
onTodayClick|func|&ensp;|true|&ensp;
title|string|&ensp;|true|&ensp;
displayToday|bool|&ensp;|true|&ensp;
todayLabel|string|&ensp;|true|&ensp;
prefix|string|&ensp;|true|&ensp;