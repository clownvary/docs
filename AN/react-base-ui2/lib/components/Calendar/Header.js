'use strict';Object.defineProperty(exports, "__esModule", { value: true });var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);var _createClass2 = require('babel-runtime/helpers/createClass');var _createClass3 = _interopRequireDefault(_createClass2);var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);var _inherits2 = require('babel-runtime/helpers/inherits');var _inherits3 = _interopRequireDefault(_inherits2);var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _propTypes = require('prop-types');var _propTypes2 = _interopRequireDefault(_propTypes);

var _consts = require('../../consts');
var _utils = require('../../utils');function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

var HeaderPropTypes = {
  onPrevClick: _propTypes2.default.func.isRequired,
  onNextClick: _propTypes2.default.func.isRequired,
  onTitleClick: _propTypes2.default.func.isRequired,
  onTodayClick: _propTypes2.default.func.isRequired,

  title: _propTypes2.default.string.isRequired,
  displayToday: _propTypes2.default.bool.isRequired,
  todayLabel: _propTypes2.default.string.isRequired,
  prefix: _propTypes2.default.string.isRequired };var


Header = function (_React$PureComponent) {(0, _inherits3.default)(Header, _React$PureComponent);function Header() {(0, _classCallCheck3.default)(this, Header);return (0, _possibleConstructorReturn3.default)(this, (Header.__proto__ || (0, _getPrototypeOf2.default)(Header)).apply(this, arguments));}(0, _createClass3.default)(Header, [{ key: 'render', value: function render()



    {var _this2 = this;
      return (
        _react2.default.createElement('div', { className: this.props.prefix + 'calendar-header' },
          _react2.default.createElement('i', {
            className: 'icon icon-chevron-left',
            onClick: this.props.onPrevClick,
            tabIndex: 0,
            onKeyDown: function onKeyDown(e) {return (0, _utils.listenKeyDown)(e,
              [_consts.KeyCode.ENTER, _consts.KeyCode.SPACE],
              function () {return _this2.props.onPrevClick();});} }),



          this.props.displayToday &&
          _react2.default.createElement('span', {
              className: this.props.prefix + 'calendar-header-today',
              onClick: this.props.onTodayClick,
              tabIndex: 0,
              onKeyDown: function onKeyDown(e) {return (0, _utils.listenKeyDown)(e,
                [_consts.KeyCode.ENTER, _consts.KeyCode.SPACE],
                function () {return _this2.props.onTodayClick();});} },

            this.props.todayLabel),

          _react2.default.createElement('span', {
              className: this.props.prefix + 'calendar-header-title',
              onClick: this.props.onTitleClick,
              tabIndex: 0,
              onKeyDown: function onKeyDown(e) {return (0, _utils.listenKeyDown)(e,
                [_consts.KeyCode.ENTER, _consts.KeyCode.SPACE],
                function () {return _this2.props.onTitleClick();});} },

            this.props.title),
          _react2.default.createElement('i', {
            className: 'icon icon-chevron-right',
            onClick: this.props.onNextClick,
            tabIndex: 0,
            onKeyDown: function onKeyDown(e) {return (0, _utils.listenKeyDown)(e,
              [_consts.KeyCode.ENTER, _consts.KeyCode.SPACE],
              function () {return _this2.props.onNextClick();});} })));




    } }]);return Header;}(_react2.default.PureComponent);Header.displayName = 'Header';Header.propTypes = HeaderPropTypes;exports.default =


Header;module.exports = exports['default'];