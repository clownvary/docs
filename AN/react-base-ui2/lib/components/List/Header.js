'use strict';Object.defineProperty(exports, "__esModule", { value: true });var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);var _createClass2 = require('babel-runtime/helpers/createClass');var _createClass3 = _interopRequireDefault(_createClass2);var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);var _inherits2 = require('babel-runtime/helpers/inherits');var _inherits3 = _interopRequireDefault(_inherits2);var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _propTypes = require('prop-types');var _propTypes2 = _interopRequireDefault(_propTypes);
var _classnames = require('classnames');var _classnames2 = _interopRequireDefault(_classnames);
var _isFunction = require('lodash/isFunction');var _isFunction2 = _interopRequireDefault(_isFunction);

var _Input = require('../Input');var _Input2 = _interopRequireDefault(_Input);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

var HeaderPropTypes = {
  config: _propTypes2.default.shape({
    disabled: _propTypes2.default.bool,
    filterable: _propTypes2.default.bool // can be filter by inputted value
  }),

  onFilter: _propTypes2.default.func };var


Header = function (_React$PureComponent) {(0, _inherits3.default)(Header, _React$PureComponent);function Header() {(0, _classCallCheck3.default)(this, Header);return (0, _possibleConstructorReturn3.default)(this, (Header.__proto__ || (0, _getPrototypeOf2.default)(Header)).apply(this, arguments));}(0, _createClass3.default)(Header, [{ key: 'renderFilter', value: function renderFilter()



    {var _props =
      this.props,_props$config = _props.config,disabled = _props$config.disabled,filterable = _props$config.filterable,onFilter = _props.onFilter;
      return filterable && (0, _isFunction2.default)(onFilter) &&
      _react2.default.createElement(_Input2.default, {
        preIcon: 'icon icon-search',
        disabled: disabled,
        onChange: function onChange(e) {return onFilter({ filter: e.target.value });} });

    } }, { key: 'render', value: function render()

    {var _props2 =
      this.props,_props2$config = _props2.config,prefix = _props2$config.prefix,filterable = _props2$config.filterable,onFilter = _props2.onFilter;

      var isHeaderEmpty = !(filterable && (0, _isFunction2.default)(onFilter));
      return (
        !isHeaderEmpty ?
        _react2.default.createElement('div', { className: (0, _classnames2.default)(prefix + 'list__header', { hidden: isHeaderEmpty }) },
          this.renderFilter()) :
        null);

    } }]);return Header;}(_react2.default.PureComponent);Header.displayName = 'Header';Header.propsType = HeaderPropTypes;exports.default =


Header;module.exports = exports['default'];