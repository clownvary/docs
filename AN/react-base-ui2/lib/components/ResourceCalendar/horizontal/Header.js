'use strict';Object.defineProperty(exports, "__esModule", { value: true });var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);var _createClass2 = require('babel-runtime/helpers/createClass');var _createClass3 = _interopRequireDefault(_createClass2);var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);var _inherits2 = require('babel-runtime/helpers/inherits');var _inherits3 = _interopRequireDefault(_inherits2);var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _classnames = require('classnames');var _classnames2 = _interopRequireDefault(_classnames);
var _isFunction = require('lodash/isFunction');var _isFunction2 = _interopRequireDefault(_isFunction);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}var

Header = function (_React$PureComponent) {(0, _inherits3.default)(Header, _React$PureComponent);function Header() {(0, _classCallCheck3.default)(this, Header);return (0, _possibleConstructorReturn3.default)(this, (Header.__proto__ || (0, _getPrototypeOf2.default)(Header)).apply(this, arguments));}(0, _createClass3.default)(Header, [{ key: 'onDateHeaderClick', value: function onDateHeaderClick(
    e, date) {
      e.stopPropagation();var

      onDateHeaderClick = this.props.onDateHeaderClick;
      if ((0, _isFunction2.default)(onDateHeaderClick)) {
        onDateHeaderClick(e, date);
      }
    } }, { key: 'renderCell', value: function renderCell(

    date) {var _this2 = this;var _props =
      this.props,_props$dateFormat = _props.dateFormat,dateFormat = _props$dateFormat === undefined ? 'DD ddd' : _props$dateFormat,currentDate = _props.currentDate,onDateHeaderClick = _props.onDateHeaderClick;
      var isToday = currentDate ? currentDate.isSame(date.value, 'day') : false;
      var isWeekend = date.value.day() === 0 || date.value.day() === 6;
      var isSunday = date.value.day() === 0;
      var classes = (0, _classnames2.default)(
      'grid-cell an-rc-date-header',
      {
        today: isToday,
        weekend: isWeekend,
        'is-sunday': isSunday });

      /* istanbul ignore next */
      var text = date ? date.value.format(dateFormat) : '';
      return (
        _react2.default.createElement('th', {
            className: classes,
            key: 'header_' + date.key },

          _react2.default.createElement('span', {
              className: (0, _classnames2.default)({ 'an-rc-clickable': (0, _isFunction2.default)(onDateHeaderClick) }),
              onClick: function onClick(e) {return _this2.onDateHeaderClick(e, date.value);} },

            isToday ? 'Today' : text)));



    } }, { key: 'render', value: function render()

    {var _this3 = this;var _props$dates =
      this.props.dates,dates = _props$dates === undefined ? [] : _props$dates;
      return (
        _react2.default.createElement('table', { className: 'an-rc-grid an-rc-grid-header' },
          _react2.default.createElement('thead', null,
            _react2.default.createElement('tr', { className: 'grid-row' },

              dates.map(function (date) {return _this3.renderCell(date);})))));





    } }]);return Header;}(_react2.default.PureComponent);exports.default =


Header;module.exports = exports['default'];