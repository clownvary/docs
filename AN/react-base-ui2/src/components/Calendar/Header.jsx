import React from 'react';
import PropTypes from 'prop-types';

import { KeyCode } from '../../consts';
import { listenKeyDown } from '../../utils';

const HeaderPropTypes = {
  onPrevClick: PropTypes.func.isRequired,
  onNextClick: PropTypes.func.isRequired,
  onTitleClick: PropTypes.func.isRequired,
  onTodayClick: PropTypes.func.isRequired,

  title: PropTypes.string.isRequired,
  displayToday: PropTypes.bool.isRequired,
  todayLabel: PropTypes.string.isRequired,
  prefix: PropTypes.string.isRequired
};

class Header extends React.PureComponent {
  static displayName = 'Header';
  static propTypes = HeaderPropTypes;

  render() {
    return (
      <div className={`${this.props.prefix}calendar-header`}>
        <i
          className="icon icon-chevron-left"
          onClick={this.props.onPrevClick}
          tabIndex={0}
          onKeyDown={e => listenKeyDown(e,
            [KeyCode.ENTER, KeyCode.SPACE],
            () => this.props.onPrevClick())
          }
        />
        {
          this.props.displayToday && (
          <span
            className={`${this.props.prefix}calendar-header-today`}
            onClick={this.props.onTodayClick}
            tabIndex={0}
            onKeyDown={e => listenKeyDown(e,
              [KeyCode.ENTER, KeyCode.SPACE],
              () => this.props.onTodayClick())
            }
          >{this.props.todayLabel}</span>)
        }
        <span
          className={`${this.props.prefix}calendar-header-title`}
          onClick={this.props.onTitleClick}
          tabIndex={0}
          onKeyDown={e => listenKeyDown(e,
            [KeyCode.ENTER, KeyCode.SPACE],
            () => this.props.onTitleClick())
          }
        >{this.props.title}</span>
        <i
          className="icon icon-chevron-right"
          onClick={this.props.onNextClick}
          tabIndex={0}
          onKeyDown={e => listenKeyDown(e,
            [KeyCode.ENTER, KeyCode.SPACE],
            () => this.props.onNextClick())
          }
        />
      </div>
    );
  }
}

export default Header;
