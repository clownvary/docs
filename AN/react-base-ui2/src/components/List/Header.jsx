import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import isFunction from 'lodash/isFunction';

import Input from '../Input';

const HeaderPropTypes = {
  config: PropTypes.shape({
    disabled: PropTypes.bool,
    filterable: PropTypes.bool // can be filter by inputted value
  }),

  onFilter: PropTypes.func
};

class Header extends React.PureComponent {
  static displayName = 'Header';
  static propsType = HeaderPropTypes;

  renderFilter() {
    const { config: { disabled, filterable }, onFilter } = this.props;
    return filterable && isFunction(onFilter) &&
      (<Input
        preIcon="icon icon-search"
        disabled={disabled}
        onChange={e => onFilter({ filter: e.target.value })}
      />);
  }

  render() {
    const { config: { prefix, filterable }, onFilter } = this.props;

    const isHeaderEmpty = !(filterable && isFunction(onFilter));
    return (
      !isHeaderEmpty ?
        <div className={classNames(`${prefix}list__header`, { hidden: isHeaderEmpty })} >
          {this.renderFilter()}
        </div> : null
    );
  }
}

export default Header;
