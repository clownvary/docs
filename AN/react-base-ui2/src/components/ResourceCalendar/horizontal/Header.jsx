import React from 'react';
import classNames from 'classnames';
import isFunction from 'lodash/isFunction';

class Header extends React.PureComponent {
  onDateHeaderClick(e, date) {
    e.stopPropagation();

    const { onDateHeaderClick } = this.props;
    if (isFunction(onDateHeaderClick)) {
      onDateHeaderClick(e, date);
    }
  }

  renderCell(date) {
    const { dateFormat = 'DD ddd', currentDate, onDateHeaderClick } = this.props;
    const isToday = currentDate ? currentDate.isSame(date.value, 'day') : false;
    const isWeekend = (date.value.day() === 0 || date.value.day() === 6);
    const isSunday = date.value.day() === 0;
    const classes = classNames(
      'grid-cell an-rc-date-header',
      {
        today: isToday,
        weekend: isWeekend,
        'is-sunday': isSunday
      });
    /* istanbul ignore next */
    const text = date ? date.value.format(dateFormat) : '';
    return (
      <th
        className={classes}
        key={`header_${date.key}`}
      >
        <span
          className={classNames({ 'an-rc-clickable': isFunction(onDateHeaderClick) })}
          onClick={e => this.onDateHeaderClick(e, date.value)}
        >
          {isToday ? 'Today' : text}
        </span>
      </th>
    );
  }

  render() {
    const { dates = [] } = this.props;
    return (
      <table className="an-rc-grid an-rc-grid-header">
        <thead>
          <tr className="grid-row">
            {
              dates.map(date => this.renderCell(date))
            }
          </tr>
        </thead>
      </table>
    );
  }
}

export default Header;
