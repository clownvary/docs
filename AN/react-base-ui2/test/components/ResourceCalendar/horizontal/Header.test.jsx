import React from 'react';
import { shallow } from 'enzyme';
import moment from 'moment-timezone';
import renderer from 'react-test-renderer';
import Header from 'src/components/ResourceCalendar/horizontal/Header';

moment.tz.setDefault('EST');
const setup = (props) => {
  const initState = {
    currentDate: moment('2018-09-19'),
    dateFormat: 'YYYY-MM-DD',
    dates: [{ value: moment('2018-09-19'), key: '2018-09-19' }, { value: moment('2018-09-13'), key: '2018-09-13' }, { value: moment('2018-09-21'), key: '2018-09-21' }]
  };
  const actions = {
    onDateHeaderClick: jest.fn()
  };
  const state = Object.assign({}, initState, props);
  const Component = shallow(<Header {...actions} {...state} />);
  const tree = renderer.create(<Header {...actions} {...state} />).toJSON();

  return { Component, actions, tree };
};

describe('src/components/ResourceCalendar/horizontal/Header', () => {
  it('should render well', () => {
    const { tree } = setup();
    expect(tree).toMatchSnapshot();
  });
  it('should render well when dates is null', () => {
    const { Component } = setup({ dates: undefined });
    expect(Component.find('.grid-cell').length).toEqual(0);
  });
  it('should render well when dateFormat is null', () => {
    const { Component } = setup({ dateFormat: undefined, currentDate: moment('2018-09-21') });
    expect(Component.find('.an-rc-clickable').first().text()).toEqual('19 Wed');
  });
  it('should render well when currentDate is null', () => {
    const { Component } = setup({ currentDate: undefined });
    expect(Component.find('.an-rc-clickable').first().text()).toEqual('2018-09-19');
  });
  describe('onDateHeaderClick:', () => {
    it('onDateHeaderClick should be called when onDateHeaderClick is not null', () => {
      const { Component, actions } = setup();
      const clickable = Component.find('.an-rc-clickable').first();
      const jsEvent = {
        stopPropagation: jest.fn()
      };
      clickable.simulate('click', jsEvent);
      expect(actions.onDateHeaderClick).toHaveBeenCalled();
    });
    it('onDateHeaderClick should be called when onDateHeaderClick is null', () => {
      const { Component, actions } = setup({ onDateHeaderClick: null });
      const clickable = Component.find('.an-rc-date-header').first();
      const jsEvent = {
        stopPropagation: jest.fn()
      };
      clickable.find('span').simulate('click', jsEvent);
      expect(actions.onDateHeaderClick).not.toHaveBeenCalled();
    });
  });
});
