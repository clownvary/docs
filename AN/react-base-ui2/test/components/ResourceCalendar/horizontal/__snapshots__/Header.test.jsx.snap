// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/ResourceCalendar/horizontal/Header should render well 1`] = `
<table
  className="an-rc-grid an-rc-grid-header"
>
  <thead>
    <tr
      className="grid-row"
    >
      <th
        className="grid-cell an-rc-date-header today"
      >
        <span
          className="an-rc-clickable"
          onClick={[Function]}
        >
          Today
        </span>
      </th>
      <th
        className="grid-cell an-rc-date-header"
      >
        <span
          className="an-rc-clickable"
          onClick={[Function]}
        >
          2018-09-13
        </span>
      </th>
      <th
        className="grid-cell an-rc-date-header"
      >
        <span
          className="an-rc-clickable"
          onClick={[Function]}
        >
          2018-09-21
        </span>
      </th>
    </tr>
  </thead>
</table>
`;
